docs/source/reference/config.rst

```diff
@@ -276,6 +276,18 @@ Available fields and semantics:
     # for details on deploying the NGINX ingress controller.
     ports: loadbalancer
 
+    # Attach custom metadata to Kubernetes objects created by SkyPilot
+    #
+    # Uses the same schema as Kubernetes metadata object: https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.26/#objectmeta-v1-meta
+    #
+    # Since metadata is applied to all all objects created by SkyPilot,
+    # specifying 'name' and 'namespace' fields here is not allowed.
+    custom_metadata:
+      labels:
+        mylabel: myvalue
+      annotations:
+        myannotation: myvalue
```

Comment:
Are both of the `labels` and `annotations` required? We are using `instance_tags:` for the other clouds to allow user specify custom tags for all the VMs created. Would it be possible to have a similar argument for k8s pods?

Reply:
Both are not required, they can be different or even just one can be present. Labels and annotations serve different purposes - labels are used for filtering, selecting and scheduling, while annotations are metadata that can be consumed by third party tools (e.g., `prometheus.io/scrape` annotations tells prometheus which objects to scrape). 

Given this, users may want to set labels and/or annotations, this config gives them the flexibility to do so.

Reply:
Makes sense! Maybe in the future, we can provide also provide `instance_tags` as an alias for specifying both `labels` and `annotations` with the same set of key, values.